### PR TITLE
fix: fuse_block panic

### LINKED ISSUE #12004: bug: fuse_block panic
```
mysql> create table t1(a int) row_per_block=3;
Query OK, 0 rows affected (0.11 sec)

mysql> insert into t1 select number from numbers(9);
Query OK, 9 rows affected (1.04 sec)

mysql> insert into t1 values(9),(10);
Query OK, 2 rows affected (0.17 sec)

mysql> select count() from fuse_block('default', 't1');
ERROR 1105 (HY000): PanicError. Code: 1104, Text = assertion failed: columns.iter().all(|entry|\n        match &entry.value {\n            Value::Scalar(_) => true,\n            Value::Column(c) =>\n                c.len() == num_rows && c.data_type() == entry.data_type,\n        }).
```

### PR DESCRIPTION
**Pull request by zhyass**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1. fix fuse_block panic with different row_num
2. fuse_segment support limit
3. fuse_segment support two args
```
select * from fuse_segment('db_name', 'tb_name'[, 'snapshot_id']) [limit segment_num];
```
4. read_segments with chunk size

- Closes #12004

### CODE REVIEW

File: src/query/storages/fuse/src/table_functions/fuse_segments/fuse_segment_table.rs [hunk 1]
```diff
@@ -103,17 +103,21 @@ impl Table for FuseSegmentTable {
...
-            string_literal(self.arg_database_name.as_str()),
-            string_literal(self.arg_table_name.as_str()),
-            string_literal(self.arg_snapshot_id.as_str()),
-        ]))
+        let mut args = Vec::new();
+        args.push(string_literal(self.arg_database_name.as_str()));
+        args.push(string_literal(self.arg_table_name.as_str()));
+        if self.arg_snapshot_id.is_some() {
+            args.push(string_literal(
+                self.arg_snapshot_id.clone().unwrap().as_str(),
```
Comment (SkyFan2002) [inline src/query/storages/fuse/src/table_functions/fuse_segments/fuse_segment_table.rs:111]: `If let` seems better

File: src/query/storages/fuse/src/table_functions/fuse_blocks/fuse_block.rs [hunk 2]
```diff
@@ -101,7 +102,7 @@ impl<'a> FuseBlock<'a> {
     #[async_backtrace::framed]
     async fn to_block(&self, snapshot: Arc<TableSnapshot>) -> Result<DataBlock> {
         let limit = self.limit.unwrap_or(usize::MAX);
-        let len = std::cmp::min(snapshot.segments.len(), limit);
+        let len = std::cmp::min(snapshot.summary.block_count as usize, limit);
```
Comment (SkyFan2002) [inline src/query/storages/fuse/src/table_functions/fuse_blocks/fuse_block.rs:105]: Thanks for fixing this.

**Review by SkyFan2002 — APPROVED** [review]

File: src/query/service/src/procedures/systems/fuse_segment.rs [hunk 3]
```diff
@@ -48,7 +48,12 @@ impl OneBlockProcedure for FuseSegmentProcedure {
     async fn all_data(&self, ctx: Arc<QueryContext>, args: Vec<String>) -> Result<DataBlock> {
         let database_name = args[0].clone();
```
Comment (RinChanNOWWW) [inline src/query/service/src/procedures/systems/fuse_segment.rs:49]: Can add some assertions here to check if the `args` is with a valid length.

File: src/query/storages/fuse/src/table_functions/fuse_segments/fuse_segment.rs [hunk 4]
```diff
@@ -82,39 +97,85 @@ impl<'a> FuseSegment<'a> {
...
-        let mut file_location: Vec<Vec<u8>> = Vec::with_capacity(len);
+        let limit = self.limit.unwrap_or(usize::MAX);
+        let len = std::cmp::min(segment_locations.len(), limit);
+
+        let mut format_versions = NumberColumnBuilder::with_capacity(&NumberDataType::UInt64, len);
+        let mut block_count = NumberColumnBuilder::with_capacity(&NumberDataType::UInt64, len);
+        let mut row_count = NumberColumnBuilder::with_capacity(&NumberDataType::UInt64, len);
+        let mut compressed = NumberColumnBuilder::with_capacity(&NumberDataType::UInt64, len);
+        let mut uncompressed = NumberColumnBuilder::with_capacity(&NumberDataType::UInt64, len);
+        let mut file_location = StringColumnBuilder::with_capacity(len, 0);
```
Comment (RinChanNOWWW) [inline src/query/storages/fuse/src/table_functions/fuse_segments/fuse_segment.rs:108]: Why change to this style?

**Review by RinChanNOWWW — APPROVED** [review]

**Pull request merged**